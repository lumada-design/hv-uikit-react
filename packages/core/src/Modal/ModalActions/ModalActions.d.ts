import { DialogActionsProps, StandardProps } from "@material-ui/core";

export type HvModalActionsClassKey = "root" | "spacing";

export type HvModalActionsProps = StandardProps<DialogActionsProps, HvModalActionsClassKey>;

export default function HvModalActions(props: HvModalActionsProps): JSX.Element | null;
