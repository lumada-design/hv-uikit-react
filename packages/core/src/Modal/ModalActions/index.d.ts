declare module '@hv/uikit-react-core/dist/Modal/ModalActions' {
  import React from 'react'

  class HvModalActions extends React.Component<
    HvModalActionsProps,
    any
  > {}

  export default HvModalActions

  export interface HvModalActionsProps
    extends React.HTMLAttributes<HvModalActions> {
    /**
     * A Jss Object used to override or extend the styles applied.
     */
    classes?: {
      /**
       * Style applied to the root of the component (container for the actions).
       */
      root?: string
      /**
       * Style applied to each child action.
       */
      action?: string
    }
  }
}
