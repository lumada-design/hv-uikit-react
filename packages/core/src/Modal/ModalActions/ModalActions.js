/*
 * Copyright 2019 Hitachi Vantara Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import PropTypes from "prop-types";
import MuiDialogActions from "@material-ui/core/DialogActions";

/**
 * Actions of the modal.
 *
 * @param classes
 * @param children
 * @param others
 * @returns {*}
 * @constructor
 */
const ModalActions = ({ classes, children, ...others }) => (
  <MuiDialogActions
    classes={{
      root: classes.root,
      action: classes.action
    }}
    {...others}
  >
    {children}
  </MuiDialogActions>
);

ModalActions.propTypes = {
  /**
   * A Jss Object used to override or extend the styles applied.
   */
  classes: PropTypes.shape({
    /**
     * Style applied to the root of the component (container for the actions).
     */
    root: PropTypes.string,
    /**
     * Style applied to each child action.
     */
    action: PropTypes.string
  }).isRequired,
  /**
   * Actions of the modal.
   */
  children: PropTypes.node.isRequired
};

export default ModalActions;
