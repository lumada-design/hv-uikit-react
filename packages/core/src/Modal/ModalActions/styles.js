const styles = theme => ({
  root: {
    margin: "0",
    padding: `${theme.hv.spacing.sm}px`,
    borderTop: `3px solid ${theme.hv.palette.atmosphere.atmo2}`
  },
  spacing: {
    "& > :not(:first-child)": {
      marginLeft: `${theme.hv.spacing.xs}px`
    }
  }
});

export default styles;
