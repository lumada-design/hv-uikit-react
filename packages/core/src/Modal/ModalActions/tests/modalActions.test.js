/* eslint-env jest */

import React from "react";
import { shallow } from "enzyme";

import HvProvider from "../../../Provider";
import ModalActions from "..";

describe("[v3] ModalActions withStyles", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = shallow(
      <HvProvider>
        <ModalActions>Modal Content</ModalActions>
      </HvProvider>
    );
  });

  it("should render correctly", () => {
    expect(wrapper.find(ModalActions)).toMatchSnapshot();
  });
});

describe("[v3] ModalActions Component", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = shallow(
      <HvProvider>
        <ModalActions>Modal Content</ModalActions>
      </HvProvider>
    );
  });

  it("should render correctly if opened", () => {
    expect(wrapper.find(ModalActions)).toMatchSnapshot();
  });

  it("allows external props to be added", () => {
    wrapper = shallow(
      <HvProvider>
        <ModalActions disableActionSpacing>Modal Content</ModalActions>
      </HvProvider>
    );
    expect(wrapper.find(ModalActions)).toMatchSnapshot();
  });

  it("allows external styles to be added", () => {
    wrapper = shallow(
      <HvProvider>
        <ModalActions
          classes={{
            root: "testClassRoot",
            action: "testClassAction",
          }}
        >
          Modal Content
        </ModalActions>
      </HvProvider>
    );
    expect(wrapper.find(ModalActions)).toMatchSnapshot();
  });
});
