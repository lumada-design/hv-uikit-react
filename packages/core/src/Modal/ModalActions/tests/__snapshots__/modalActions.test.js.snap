// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalActions Component allows external props to be added 1`] = `
<WithStyles(DialogActions)
  className=""
  classes={
    Object {
      "action": undefined,
      "root": undefined,
    }
  }
  disableActionSpacing={true}
>
  Modal Content
</WithStyles(DialogActions)>
`;

exports[`ModalActions Component allows external styles to be added 1`] = `
<WithStyles(DialogActions)
  className=""
  classes={
    Object {
      "action": "testClassAction",
      "root": "testClassRoot",
    }
  }
>
  Modal Content
</WithStyles(DialogActions)>
`;

exports[`ModalActions Component should render correctly if opened 1`] = `
<WithStyles(DialogActions)
  className=""
  classes={
    Object {
      "action": undefined,
      "root": undefined,
    }
  }
>
  Modal Content
</WithStyles(DialogActions)>
`;

exports[`ModalActions withStyles should render correctly 1`] = `
<JssProvider
  generateClassName={[Function]}
>
  <HvProvider
    changeTheme={[Function]}
    router={null}
    theme={null}
    uiKitTheme="dawn"
  >
    <WithStyles(ModalActions)>
      Modal Content
    </WithStyles(ModalActions)>
  </HvProvider>
</JssProvider>
`;
